vllm/spec_decode/top1_proposer.py

```diff
@@ -73,6 +73,29 @@ def get_proposals(
                 execute_model_req=nonzero_execute_model_req,
                 sample_len=proposal_len,
             )
+            if maybe_sampler_output is not None:
+                # Some sequences do not get speculative tokens from
+                # the draft worker. Remove these sequences from nonzero
+                # proposal len seqs to reduce scoring overhead
+                zero_seq_idxs = []
```

Comment:
Can we move the contents of this branch to own function for readability and separation?